effekt/shared/src/main/scala/effekt/core/PolymorphismBoxing.scala

```diff
@@ -525,15 +525,15 @@ object PolymorphismBoxing extends Phase[CoreTransformed, CoreTransformed] {
         }
 
         override def callDirect(block: B, vargs: List[Pure], bargs: List[Block])(using PContext): Expr = {
-          val result = TmpValue()
+          val result = TmpValue("callDirectResult")
           Run(Let(result, rcoercer.from, DirectApp(block, targs map transformArg,
             (vcoercers zip vargs).map {case (c,v) => c(v)},
             (bcoercers zip bargs).map {case (c,b) => c(b)}),
             Return(rcoercer(Pure.ValueVar(result, rcoercer.from)))))
         }
 
         override def call(block: B, vargs: List[Pure], bargs: List[Block])(using PContext): Stmt = {
-          val result = TmpValue()
+          val result = TmpValue("callResult")
```

Comment:
Same as above. - All the ones in boxing should probably be called the same, as they are all generated for the value before coercion.